cmd/commands/product-release.go

```diff
@@ -0,0 +1,245 @@
+	Slice struct {
+		Edges    []*Edge              `json:"edges"`
+		PageInfo *platmodel.SliceInfo `json:"pageInfo"`
+	}
+
+	Edge struct {
+		Node map[string]any `json:"node"`
+	}
+)
+
+func NewProductReleaseCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "product-release",
+		Short:             "Manage product releases of Codefresh account",
+		PersistentPreRunE: cfConfig.RequireAuthentication,
+		Args:              cobra.NoArgs, // Workaround for subcommand usage errors. See: https://github.com/spf13/cobra/issues/706
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.HelpFunc()(cmd, args)
+			exit(1)
+		},
+	}
+
+	cmd.AddCommand(NewProductReleaseListCommand())
+
+	return cmd
+}
+
+func NewProductReleaseListCommand() *cobra.Command {
+
+	var (
+		productName       string
+		page              int
+		pageLimit         int
+		statusList        string
+		promotionFlowList string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all the pipelines",
+		Args:  cobra.NoArgs,
+		Example: util.Doc(`
+			<BIN> product-release list --product <product>
+		`),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			ctx := cmd.Context()
+
+			releaseStatus, err := ToProductReleaseStatus(statusList)
+			promotionFlows, err := ToPromotionFlows(promotionFlowList)
+			if err != nil {
+				return err
+			}
+			filterArgs := platmodel.ProductReleaseFiltersArgs{
+				Statuses:       releaseStatus,
+				PromotionFlows: promotionFlows,
+			}
+			return RunProductReleaseList(ctx, filterArgs, productName, page, pageLimit)
+		},
+	}
+
+	cmd.Flags().StringVarP(&statusList, "status", "s", "", "Filter by statuses")
+	cmd.Flags().StringVar(&promotionFlowList, "promotion-flows", "", "Filter by promotion flows")
+	cmd.Flags().IntVar(&page, "page", 1, "page number")
+	cmd.Flags().IntVar(&pageLimit, "page-limit", 20, "page limit number")
+	cmd.Flags().StringVarP(&productName, "product", "p", "", "product")
+
+	return cmd
+}
+
+func RunProductReleaseList(ctx context.Context, filterArgs platmodel.ProductReleaseFiltersArgs, productName string, page int, pageLimit int) error {
+	pagination := platmodel.SlicePaginationArgs{
+		First: &pageLimit,
+	}
+
+	query := `
+	query getProductReleasesList(
+	  $productName: String!
+	  $filters: ProductReleaseFiltersArgs!
+	  $pagination: SlicePaginationArgs
+	) {
+	  productReleases(productName: $productName, filters: $filters, pagination: $pagination) {
+		pageInfo {
+			hasNextPage
+			hasPrevPage
+			startCursor
+			endCursor
+		}
+		edges {
+		  node {
+			releaseId
+			steps {
+			  environmentName
+			  status
+			}
+			status
+		  }
+		}
+	  }
+	}
+  `
+	// add pagination - default for now is last 20
+	variables := map[string]any{
+		"filters":     filterArgs,
+		"productName": productName,
+		"pagination":  pagination,
+	}
+
+	var productReleasesPage map[string]any
+	productReleasesPage, err := client.GraphqlAPI[map[string]any](ctx, cfConfig.NewClient().InternalClient(), query, variables)
```

Comment:
i think that will do
```suggestion
	productReleasesPage, err := client.GraphqlAPI[map[string]any](ctx, cfConfig.NewClient().InternalClient(), query, variables)
```

Reply:
also - shouldn't this just return the `Slice`?